Add ability to set portHeader in tomcat's server.xml

tareqmamari opened the pull request:
If UAA receives a request from a proxy/load balancer where a port mapping is happening, any redirect from UAA will pick the default port number (443 if https), which is the wrong port where that proxy/load balancer is listening to and doing the ssl termination. This commit updates the tomcat 
server config file `server.xml` to get the port from `X-Forwarded-Port` if it is specified. This header should be set by the proxy/load balancer.

tareqmamari committed:
Add ability to set portHeader in tomcat's server.xml

If UAA receives a request from a proxy/load balancer where a port mapping is happening,
any redirect from UAA will pick the default port number (443 if https), which is the wrong
port where that proxy/load balancer is listening to. This commit updates the tomcat server
config file `server.xml` to get the port from `X-Forwarded-Port` if it is specified. This header
should be set by the proxy/load balancer.

tareqmamari commented:
Is anyone looking at this please ?

shamus commented:
I see you've rebased! Thanks. 

But it looks like you haven't run the tests, they're failing. I see this step is missing from the README, I'll add it.

If you could run the test suite and ensure the tests pass that would be best. It's an rspec test suite so you'll need ruby. To run them:

```
gem install bundler
bundle install
bundle exec rake
```

tareqmamari committed:
Add test coverage for setting portHeader in tomcat

This is to add test coverage for the commit https://git.io/Je5lk
which will make sure setHeader property is added to the expected tomcat's
server config

tareqmamari commented:
Thanks @shamus, it should be OK now
